apps/web/src/pages/ApplicantDashboardPage/components/ReviewRecruitmentProcessPreviewList.tsx

```diff
@@ -0,0 +1,557 @@
+                      },
+                    ]}
+                  />
+                  <Dialog.Footer data-h2-gap="base(0 x1)">
+                    <Button
+                      type="submit"
+                      disabled={isSubmitting}
+                      color="secondary"
+                    >
+                      {intl.formatMessage(formMessages.saveChanges)}
+                    </Button>
+                    <Link
+                      href={paths.application(recruitmentProcess.id)}
+                      mode="inline"
+                      color="secondary"
+                    >
+                      {intl.formatMessage({
+                        defaultMessage: "View application",
+                        id: "xg/wvH",
+                        description: "Label for view application link",
+                      })}
+                    </Link>
+                    <Link
+                      href={paths.pool(pool.id)}
+                      mode="inline"
+                      color="secondary"
+                    >
+                      {intl.formatMessage({
+                        defaultMessage: "View job advertisement",
+                        id: "eZlUrp",
+                        description: "Label for view job advertisement link",
+                      })}
+                    </Link>
+                    <Dialog.Close>
+                      <Button mode="inline" color="warning">
+                        {intl.formatMessage(commonMessages.cancel)}
+                      </Button>
+                    </Dialog.Close>
+                  </Dialog.Footer>
+                </form>
+              </FormProvider>
+            </>
+          ) : (
+            <Dialog.Footer data-h2-gap="base(0 x1)">
+              <Dialog.Close>
+                <Button mode="solid" color="secondary">
+                  {intl.formatMessage(commonMessages.close)}
+                </Button>
+              </Dialog.Close>
+              <Link
+                href={paths.application(recruitmentProcess.id)}
+                mode="inline"
+                color="secondary"
+              >
+                {intl.formatMessage({
+                  defaultMessage: "View application",
+                  id: "xg/wvH",
+                  description: "Label for view application link",
+                })}
+              </Link>
+              <Link href={paths.pool(pool.id)} mode="inline" color="secondary">
+                {intl.formatMessage({
+                  defaultMessage: "View job advertisement",
+                  id: "eZlUrp",
+                  description: "Label for view job advertisement link",
+                })}
+              </Link>
+            </Dialog.Footer>
+          )}
+        </Dialog.Body>
+      </Dialog.Content>
+    </Dialog.Root>
+  );
+};
+
+const ReviewRecruitmentProcessPreviewList = ({
+  recruitmentProcesses,
+}: {
+  recruitmentProcesses: ReviewRecruitmentProcessPreviewListFragment[];
+}) => {
```

Comment:
Two things. Can this be an interface and secondly, I think it should be using the `FragmentType` generic.

Reply:
Changed in [b7e4eb3](https://github.com/GCTC-NTGC/gc-digital-talent/pull/12366/commits/b7e4eb30269e46bbc3948d9cbf267c7c15a45111)